# 2. Does participating in these directed reading activities in the classroom help elementary school students improve aspects of their reading ability? {-}

## Our data (from __[Here](http://userhome.brooklyn.cuny.edu/cbenes/Reading.txt)__):{-}
x is a set of Degree of Reading Power(DRP) test scores from a class of 21 third-grade students who participated in the directed reading activities for eight weeks.
y is a set of DRP test scores from a class of 23 third-grade students who did not participate in these activities.

In [64]:
www = "http://userhome.brooklyn.cuny.edu/cbenes/Reading.txt"
TIMES = read.table(www)
x = TIMES[1:21, 2] 
y = TIMES[22:44, 2]
n = length(x)
m = length(y)

##  1. Test if the data set x and y have the same variances{-}
### Assumptions we are making:{-}
The both x and y are independent random samples from normal distributions with means $\mu_x$ and $\mu_y$ and standard deviations $\sigma_x$ and $\sigma_y$ respectively.


### The hypotheses to be tested are{-}
\begin{align}
H_0 : \sigma^2_x = \sigma^2_y\\
versus \quad \\
H_1 : \sigma^2_x \neq \sigma^2_y\\
\end{align}


In [65]:
sum_x = sum(x)
sum_x
sum_x_sqrd = sum(x^2)
sum_x_sqrd
sum_y = sum(y)
sum_y
sum_y_sqrd = sum(y^2)
sum_y_sqrd

[1] 1081

[1] 58069

[1] 955

[1] 46123

### Let alpha = 0.05 be the level of significance. {-}
\begin{align}
\sum_{i=1}^{21} x_i =  1081 \quad \quad \sum_{i=1}^{21} x^2_i =  58069\\
\sum_{i=1}^{23} y_i =  955 \quad \quad \sum_{i=1}^{23} y^2_i =  46123\\
\end{align}



In [66]:
sample_var_x = (n * sum_x_sqrd - (sum_x)^2) / (n * (n-1))
sample_var_x 
sample_var_y = (m * sum_y_sqrd - (sum_y)^2) / (m * (m-1))
sample_var_y 

[1] 121.1619

[1] 294.0791

### The sample variances become {-}
\begin{align}
s^2_x = 121.161904761905\\
s^2_y = 294.079051383399\\
\end{align}


In [67]:
f_observed = sample_var_y / sample_var_x
f_observed

[1] 2.427158

### Dividing the sample variances gives an observed F ratio of 2.4271577106789 {-}
\begin{align}
F =  \frac{s^2_y}{s^2_x} = \frac{294.079051383399}{121.161904761905} = 2.4271577106789 \\
\end{align}

### Our decision rule:{-}
We reject $H_0$ if $\frac{s^2_y}{s^2_x}$ is either $(1) \leq F_{0.025,22,20}$ or $(2) \geq F_{0.975,22,20}$

In [68]:
f_1 = qf(0.025, df1 = 22, df2 = 20)
f_1
f_2 = qf(0.975, df1 = 22, df2 = 20)
f_2

[1] 0.4185882

[1] 2.433673

### We have {-}
\begin{align}
F_{0.025,22,20} = 0.41858818655092 \\
F_{0.975,22,20} = 2.43367309113588 \\
\end{align}
### Then,{-}
\begin{align}
0.41858818655092 < 2.4271577106789 < 2.43367309113588 \\
\end{align}

### Conclusion: {-}
Therefore, we fail to reject our null hypothesis, suggesting that we don't have enough evidence to claim that the variances for the two data sets are different.

##  2. Test if the data set x and y have the same mean{-}
### Assumptions we are making:{-}
Part 1 allows us to assume that $\sigma_x = \sigma_y$, so we can use the two-sample t-test. Also, it is reasonable to use a one-sided test here because if there is any effect at all, participating in the the directed reading activities in the classroom will improve one's reading ability.
<br>

### The hypotheses to be tested are{-}
\begin{align}
H_0 : \mu_x = \mu_y\\
versus \quad \\
H_1 : \mu_x > \mu_y\\
\end{align}

### We reject $H_0$ if{-}
\begin{align}
t = \frac{\bar{x} - \bar{y}}{S_p\sqrt{\frac{1}{n} + \frac{1}{m}}} \leq -t_{0.05, 42}\\
\end{align}
<br>
Since we already calculated the sample variances in part 1, now we just need the sample means $\bar{x}$ and $\bar{y}$ and the pooled standard deviation $S_p$

In [69]:
sample_mean_x = mean(x)
sample_mean_x
sample_mean_y = mean(y)
sample_mean_y
s_p = sqrt(((n-1) * sample_var_x + (m-1) * sample_var_y) / (n+m-2))
s_p

[1] 51.47619

[1] 41.52174

[1] 14.5512

### We have{-}
\begin{align}
\bar{x} = 51.4761904761905 \\
\bar{y} = 41.5217391304348 \\
S_p = 14.551204520321\\
\end{align}


In [70]:
t_obs = (sample_mean_x - sample_mean_y)/(s_p * sqrt((1/n) + (1/m)))
t_obs
t_score = qt(0.05, n + m - 2)
t_score

[1] 2.266552

[1] -1.681952

### Therefore{-}
\begin{align}
t = 2.26655159958594 > -1.68195235746753 = t_{0.05, 42}\\
\end{align}

### Conclusion:{-}
So there is sufficient evidence to reject the null hypothesis in favor of the alternate hypothesis. In other words, there is strong evidence that students who participated in the directed reading activities in clasroom yields better scores on the DRP test than those who did not. 